server.js

```diff
@@ -13,16 +14,100 @@ mongoose.Promise = Promise
 //   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
+const listEndpoints = require('express-list-endpoints')
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
```

Comment:
Maybe a comment around what this is doing?

Reply:
Comment added `//middleware f() -> checks if we have a connection with the server (1 = connected)`